lib/hydra/works/models/concerns/generic_file/characterization.rb

```diff
@@ -0,0 +1,120 @@
+include RDF
+
+module Hydra::Works::GenericFile
+  module Characterization
+    extend ActiveSupport::Concern
+
+    autoload :FitsDatastream, 'hydra/works/models/generic_file/characterization/fits_datastream.rb'
+    autoload :Image, 'hydra/works/models/concerns/generic_file/characterization/image.rb'
+    autoload :Doc, 'hydra/works/models/concerns/generic_file/characterization/doc.rb'
+    autoload :Video, 'hydra/works/models/concerns/generic_file/characterization/video.rb'
+    autoload :Audio, 'hydra/works/models/concerns/generic_file/characterization/audio.rb'
+
+    included do
+      include Hydra::Derivatives::ExtractMetadata
+
+      # Base characterization properties
+
+      property :filename, predicate: RDF::Vocab::EBUCore.filename, multiple: false
+      property :format_label, predicate: RDF::Vocab::PREMIS.hasFormatName
+      property :file_size, predicate: RDF::Vocab::EBUCore.fileSize
+      property :well_formed, predicate: RDF::URI.new("http://projecthydra.org/ns/fits/wellFormed")
+      property :valid, predicate: RDF::URI.new("http://projecthydra.org/ns/fits/valid")
+      property :date_created, predicate: RDF::Vocab::EBUCore.dateCreated
+      property :last_modified, predicate: RDF::Vocab::EBUCore.dateModified
+      property :fits_version, predicate: RDF::Vocab::PREMIS.hasCreatingApplicationVersion
+      property :exif_version, predicate: EXIF.exifVersion
+      property :original_checksum, predicate: RDF::Vocab::PREMIS.hasMessageDigest
+      # property :mime_type, delegate_to: 'original_file'
```

Comment:
:OK: Can you include that in a comment above that commented-out code?

On Fri, Sep 11, 2015 at 11:04 AM Colin <notifications@github.com> wrote:

> In lib/hydra/works/models/concerns/generic_file/characterization.rb
> <https://github.com/projecthydra-labs/hydra-works/pull/209#discussion_r39298727>
> :
>
> > +    included do
> > +      include Hydra::Derivatives::ExtractMetadata
> > +
> > +      # Base characterization properties
> > +
> > +      property :filename, predicate: RDF::Vocab::EBUCore.filename, multiple: false
> > +      property :format_label, predicate: RDF::Vocab::PREMIS.hasFormatName
> > +      property :file_size, predicate: RDF::Vocab::EBUCore.fileSize
> > +      property :well_formed, predicate: RDF::URI.new("http://projecthydra.org/ns/fits/wellFormed")
> > +      property :valid, predicate: RDF::URI.new("http://projecthydra.org/ns/fits/valid")
> > +      property :date_created, predicate: RDF::Vocab::EBUCore.dateCreated
> > +      property :last_modified, predicate: RDF::Vocab::EBUCore.dateModified
> > +      property :fits_version, predicate: RDF::Vocab::PREMIS.
>
> hasCreatingApplicationVersion
> > +      property :exif_version, predicate: EXIF.exifVersion
> > +      property :original_checksum, predicate: RDF::Vocab::PREMIS.hasMessageDigest
> > +      # property :mime_type, delegate_to: 'original_file'
>
> This is actually a TODO. The original_file.mime_type can currently be
> different than the generic_file.mime_type but I couldn't sort out the
> syntax for delegate_to: that would get that working.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/projecthydra-labs/hydra-works/pull/209/files#r39298727>
> .
>